pyproject.toml

```diff
@@ -0,0 +1,42 @@
+[build-system]
+requires = [
+    "setuptools==65.5.1",
+    "setuptools_scm[toml]==7.0.5",
+]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+
+[project]
+name = "bug-master-bot"
+description = "Slack bot for handling PROW failures on slack CI channels"
+readme = "README.md"
+license = { file = "LICENSE" }
+classifiers = ["Programming Language :: Python :: 3"]
+dependencies = [
+    "uvicorn==0.22.0",
```

Comment:
by convention ``pyproject.toml`` have non-pinned dependencies and ``requirements.txt`` have pinned dependencies

bug-master-bot is a service, so I think it makes more sense to work with ``requirements.txt`` and have the dependencies pinned. maybe later on we can also let dependabot update the dependencies automatically for us

Reply:
> by convention pyproject.toml have non-pinned dependencies and requirements.txt have pinned dependencies

is there a reason for that?

Reply:
yes, because when you install the package as a dependency (e.g. ``pip install bug-master``) with pinned dependencies it becomes ultra hard to integrate with other packages in the same virtual-env. So with less strict versions it makes it much more possible
``requirements.txt`` is usually self-used in the context of the component itself

Reply:
> yes, because when you install the package as a dependency (e.g. pip install bug-master) with pinned dependencies it becomes ultra hard to integrate with other packages in the same virtual-env

ok makes sense, thanks!

Reply:
@danmanor maybe I wasn't clear
``requirements.txt`` have all the deps, all pinned. In our case you can also delete all the requirements from ``pyproject.toml``, because it won't be used externally

Reply:
thanks